specs/src/specs/data_structures.md

```diff
@@ -482,28 +482,48 @@ Finally, the `availableDataRoot` of the block [Header](#header) is computed as t
+  - The first 7 bits represent the share version in big endian form (initially, this will be `0000000` for version `0`);
+  - The last bit is a sequence start indicator, that is `1` if the share is at the start of a sequence or `0` if it is a continuation share.
+
+The remainder of a share's raw data `rawData` is interpreted differently depending on the namespace ID.
+
+#### Compact Share
 
 For shares **with a reserved namespace ID through [`NAMESPACE_ID_MAX_RESERVED`](./consensus.md#constants)**:
 
-- The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data `rawData` is the namespace ID of that share, `namespaceID`.
-- The next [`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes (the `*` in the example layout figure below) is the starting byte of the length of the [canonically serialized](#serialization) first request that starts in the share, or `0` if there is none, as a one-byte big-endian unsigned integer (i.e. canonical serialization is not used). In this example, with a share size of `256` the first byte would be `80` (or `0x50` in hex).
-- The remaining [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_ID_BYTES`](./consensus.md#constants)`-`[`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes are request data.
+- If this is the first share of a sequence, the next 1-10 bytes contain a [varint](https://developers.google.com/protocol-buffers/docs/encoding) of the uint64 length of the sequence that follows.
+- The next [`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes is the starting byte of the length of the [canonically serialized](#serialization) first request that starts in the share, or `0` if there is none, as a one-byte big-endian unsigned integer (i.e. canonical serialization is not used). In the example below, with a share size of `256` the reserved byte would be `80` (or `0x50` in hex).
+- The remaining bytes are transactions, intermediate state roots, or evidence data depending on the namespace of ths share. Each transaction, intermediate state root, or evidence is prefixed with a [varint](https://developers.google.com/protocol-buffers/docs/encoding) of the length of that unit.
+- If there is insufficient transaction, intermediate state root, or evidence data to fill the share, the remaining bytes are filled with `0`.
+
+First share in a sequence:
+![fig: compact start share.](./figures/compact_start_share.svg)
+
+Continuation share in a sequence:
+![fig: compact continuation share.](./figures/compact_continuation_share.svg)
 
-![fig: Reserved share.](./figures/share.svg)
+#### Sparse Share
 
 For shares **with a namespace ID above [`NAMESPACE_ID_MAX_RESERVED`](./consensus.md#constants) but below [`PARITY_SHARE_NAMESPACE_ID`](./consensus.md#constants)**:
 
-- The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data `rawData` is the namespace ID of that share, `namespaceID`.
```

Comment:
Can you convert this into a note, if it's still valid? E.g.

> **Note** The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data `rawData` is the namespace ID of that share, `namespaceID`.

Reply:
Yep it is still valid. If we include this note here, it seems relevant to also describe
1. The share info byte: https://github.com/rootulp/celestia-app/blob/40dc16bd4e9c792d634ff65b70bdba5c2abc99a4/specs/src/specs/data_structures.md?plain=1#L486-L488
1. To include the same note in Sparse Share section below.

I opted to avoid the duplication but I'll push a commit shortly that addresses this ^ and if it seems redundant we can remove.

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/856/commits/350fb2e9fdf286f5e5761860938c984ec3d60838